pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -222,6 +235,19 @@ func (c *Cluster) createDeviceSetPVC(existingPVCs map[string]*v1.PersistentVolum
 	if existingPVC != nil {
 		logger.Infof("OSD PVC %q already exists", existingPVC.Name)
 
+		desiredPvcSize, desiredOK := pvc.Spec.Resources.Requests[v1.ResourceStorage]
+		actualPvcSize, currentOK := existingPVC.Spec.Resources.Requests[v1.ResourceStorage]
+		if !desiredOK || !currentOK {
+			logger.Debugf("desired or current size are not specified for PVC %q", existingPVC.Name)
+			return existingPVC, nil
+		}
+
+		if desiredPvcSize.Cmp(actualPvcSize) != 0 {
+			pvcResizeMap[existingPVC.Name] = pvcResize{
+				desiredSize: desiredPvcSize,
+				actualSize:  actualPvcSize,
+			}
+		}
 		// Update the PVC in case the size changed
 		k8sutil.ExpandPVCIfRequired(c.clusterInfo.Context, c.context.Client, pvc, existingPVC)
```

Comment:
How about changing the `ExpandPVCIfRequired()` to return a bool indicating if the PVC was resized. Then the decision about resizing can be owned by that method, and no need to have the separate detection code above on lines 238-250.
```suggestion
		resized := k8sutil.ExpandPVCIfRequired(c.clusterInfo.Context, c.context.Client, pvc, existingPVC)
		if resized {
			pvcResizeMap[existingPVC.Name] = pvcResize{
				desiredSize: desiredPvcSize,
				actualSize:  actualPvcSize,
			}
		}
```

Reply:
we still need some instance of it